addons/account/tests/test_fiscal_position.py

```diff
@@ -0,0 +1,31 @@
+from openerp.tests.common import TransactionCase
+
+class TestFiscalPosition(TransactionCase):
+    """Tests for fiscal positions (account.fiscal.position).
+    """
+
+    def setUp(self):
+        super(TestFiscalPosition, self).setUp()
+        self.fiscal_position_model = self.registry('account.fiscal.position')
+        self.res_partner_model = self.registry('res.partner')
+        self.res_country_model = self.registry('res.country')
+
+    def test_fiscal_position(self):
+        cr, uid = self.cr, self.uid
+        company_id = 1
+        country_id = self.res_country_model.search(cr, uid, [('name', '=', 'France')])[0]
+        partner_id = self.res_partner_model.create(cr, uid, dict(
+                                                   name="George",
+                                                   vat_subjected=True,
+                                                   notify_email="always",
+                                                   country_id=country_id))
+        fp1_id = self.fiscal_position_model.create(cr, uid, dict(name="EU-VAT-FR",
+                                                                 auto_apply=True,
+                                                                 country_id=country_id,
+                                                                 vat_required=False))
+        fp2_id = self.fiscal_position_model.create(cr, uid, dict(name="EU-VAT-FR",
+                                                                 auto_apply=True,
+                                                                 country_id=country_id,
+                                                                 vat_required=True))
+        res = self.fiscal_position_model.get_fiscal_position(cr, uid, company_id, partner_id)
+        self.assertEquals(fp2_id, res)
```

Comment:
As a best practice, each `assert*` should have an error message to be displayed when the assertion fails. It serves as a documentation of the test. Here you could have:
```
self.assertEquals(fp2_id, res, "Fiscal position detection should pick B2C position as 1rst match")
```
Using more descriptive names for the variables would also help make the test self-explaining: `fp1_id` -> `fp_b2c_id`, `fp2_id` -> `fp_b2b_id`

Finally, it would make sense to show that the detection is not broken the other way around: add another test step that disables the auto-apply mode for the B2B position, and check that now that `get_fiscal_position()` indeed returns the B2C position as fallback.